acra/acra-in-depth/security-design/_index.md

```diff
@@ -2,3 +2,165 @@
+
+The most dangerous current known security threats are:
+- Threats related to excess / abuse of access privileges.
+- Data leaks caused by mistakes in deploy, configuration settings, backup thefts, etc.
+- SQL injections.
+- Denial of service.
+- Vulnerabilities in the database protocol.
+- Weak audit or an absence thereof.
+- Operational system vulnerabilities.
+- Unsafe handling of cryptographic keys.
+
+### Conditions of secure work
+Acra can perform its protective functions properly and protect from the security threats 1, 2, 3 if the following security assumptions are met:
+- The PKI infrastructure is trusted;
+- AcraServer is trusted;
+- The client is less trusted than the server.
+
+### Possible consequences of compromisation
+Let’s consider all the possible consequences of any of separate component being broken (broken as in “fully compromised” when the adversary fully overtakes the work of the component and gains full access to its memory).
+
+When a *Database* is broken into, the worst-case scenario is DoS or COA. Thus, the stability of the system, in this case, is reduced to the stability of the symmetric encryption algorithm (AES-GCM-256).
+When the *Client* gets broken, the worst-case scenario is that the adversary can get the data belonging to this client, which is stored in the database.
+And finally, if AcraServer gets broken, the adversary can fully compromise the system.
+
+It is worth mentioning that in absence of PKI, the communication channel between the Client and AcraServer is also vulnerable. In this case, the resistance ability of the system comes down to the secureness of the SSL/TLS or Themis’ Secure Session protocols. In all the other communication channels the data is encrypted so, in the worst case (when SSL/TLS is not used) the secureness of the system comes down to the secureness of the symmetric encryption algorithm (AES-GCM-256).
+
+### Additional reading
+
+We recommend that you also check out the following articles to gain a better understanding of the security notions in this article: <br>
+https://www.zdnet.com/article/the-top-ten-most-common-database-security-vulnerabilities/ <br>   
+https://en.wikipedia.org/wiki/Vulnerability_database <br>           
+https://www.bcs.org/content/ConWebDoc/8852. <br>
+
+
+## PKI overview for Acra
+
+We strongly recommend following the best security practices and using the public key infrastructure (PKI) to provide authentication between the components of Acra. This document contains some useful information about the PKI and links to related resources.
+
+### The Public Key Infrastructure
+
+The central component of any PKI is the Certificate Authority (CA). The CA acts as an arbitrator or a third party, whilst the trust relations are established between the two other sides. A very simplified scheme of one-side authentication of `User` by a `Relying Party` is described below.
+
+
+
+![](/files/wiki/pki-overview-for-acra.png)
+
```

Comment:
Same thing as elsewhere about placing files under `files/wiki` and about `{{ ref }}` usage.